supervision/detection/smoother.py

```diff
@@ -0,0 +1,172 @@
+from collections import defaultdict
+
+import numpy as np
+
+from supervision.detection.core import Detections
+
+
+class Smoother:
+    """
+    A class for smoothing out noise in predictions over time by using a Tracker
+    to track objects over time and averaging out the predictions over the
+    `length` most recent frames.
+
+    <video controls>
+        <source src="https://media.roboflow.com/supervision/video-examples/smoothed-grocery-example-720.mp4" type="video/mp4">
+    </video>
+    > _On the left are the model's raw predictions,
+    > on the right is the output of Smoother._
+
+    ## Example Usage:
```

Comment:
Can we show a more general example? I assume `Smoother` can work with any object detection model.

Reply:
Sure, what do you have in mind? I figured person detection with COCO using one of the videos from `assets` was about as general as it gets to demonstrate the functionality but we can change it to something else.

Good candidate videos to communicate the functionality would have
* A small number of objects moving fairly subtly (otherwise it's really distracting trying to understand what you're supposed to be comparing side by side)
* Prediction jitter coming from the model

For example, while it **works** with [vehicles-2.mp4](https://media.roboflow.com/supervision/video-examples/vehicles-2.mp4), it's not a good example video because there's too much going on and the things are moving too fast to be able to grok what's going on at the detection by detection level.